newrelic-agent/src/main/java/com/newrelic/agent/config/SystemPropertyProvider.java

```diff
@@ -104,9 +85,10 @@ private void addNewRelicEnvProperties(Map<String, Object> nrProps, Map<String, S
         for (Map.Entry<String, String> entry : allEnvVars.entrySet()) {
             String envVar = entry.getKey();
             if (envVar.startsWith(NEW_RELIC_PREFIX_ENV)) {
-                String envVarNameToReplace = envVarToSystemPropKeyMap.get(envVar);
-                if (envVarNameToReplace != null) {
-                    addPropertyWithoutSystemPropRoot(nrProps, envVarNameToReplace, entry.getValue());
+                if (envVar.equals(LOG_ENV)) {
```

Comment:
This block of code has always had a sorting issue; if both `NEW_RELIC_LOG` and `NEW_RELIC_LOG_FILE_NAME` are present and different, one or the other will be chosen. I opted not to try and fix this.